File: src/main/java/szewek/flux/util/Metal.java
package szewek.flux.util;

import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import szewek.flux.item.MetalItem;

public enum Metal {
	IRON(0xE2C0BA, "iron", 1),
	GOLD(0xFFD700, "gold", 2),
	COPPER(0xE99868, "copper", 1),
	TIN(0xD0D3D6, "tin", 1);
	//BRONZE(0x000, "bronze", 2);

	public final int color;
	public final String metalName;
	public final int harvestLevel;

	Metal(int color, String metalName, int harvestLevel) {
		this.color = color;
		this.metalName = metalName;
		this.harvestLevel = harvestLevel;
	}

	public final boolean nonVanilla() {
		return this != IRON && this != GOLD;
	}

	public final boolean nonAlloy() {
		return true;//return this != BRONZE;
	}

	public final boolean notVanillaOrAlloy() {
		return nonVanilla() && nonAlloy();
	}

	public final boolean all() {
		return true;
	}

	private static int getColor(ItemStack stack) {
		Item item = stack.getItem();
		return item instanceof MetalItem ? ((MetalItem)item).metal.color : 0;
	}

	public static int gritColors(ItemStack stack, int layer) {
		return layer == 0 ? 0xffffff : getColor(stack);
	}

	public static int ingotColors(ItemStack stack, int layer) {
		return layer != 0 ? 0xffffff : getColor(stack);
	}

	public static int itemColors(ItemStack stack, int layer) {
		return getColor(stack);
	}
}

File: src/main/java/szewek/flux/util/Metal.kt
package szewek.flux.util

import net.minecraft.item.ItemStack
import szewek.flux.item.MetalItem

enum class Metal(
        @JvmField val color: Int,
        @JvmField val metalName: String,
        @JvmField val harvestLevel: Int
) {
    IRON(0xE2C0BA, "iron", 1), GOLD(0xFFD700, "gold", 2), COPPER(0xE99868, "copper", 1), TIN(0xD0D3D6, "tin", 2);

    fun always() = true

    fun nonVanilla() = this != IRON && this != GOLD

    companion object {
        private fun getColor(stack: ItemStack): Int {
            val item = stack.item
            return if (item is MetalItem) {
                item.metal!!.color
            } else 0
        }

        @JvmStatic
        fun gritColors(stack: ItemStack, layer: Int): Int {
            return if (layer == 0) 0xFFFFFF else getColor(stack)
        }

        @JvmStatic
        fun ingotColors(stack: ItemStack, layer: Int): Int {
            return if (layer != 0) 0xFFFFFF else getColor(stack)
        }

        @JvmStatic
        fun itemColors(stack: ItemStack, layer: Int): Int {
            return getColor(stack)
        }
    }

}